packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+const converterKeysTableName = getEnvOrFail("FHIR_T0_CDA_CONVERTER_KEYS_TABLE_NAME");
+
+const lambdaClient = makeLambdaClient(region);
+const s3Client = makeS3Client(region);
+
+export const handler = Sentry.AWSLambda.wrapHandler(
+  async (req: {
+    bundle: Bundle<Resource>;
+    patientId: string;
+    cxId: string;
+    resources?: ResourceTypeForConsolidation[];
+    dateFrom?: string;
+    dateTo?: string;
+    conversionType: string;
+  }) => {
+    const { bundle, patientId, cxId, resources, dateFrom, dateTo, conversionType } = req;
+
+    try {
+      const log = prefixedLog(`patient ${patientId}`);
+      const res = await postToConverter({
+        url: FHIRToCDAConverterUrl,
+        payload: bundle,
+        converterKeysTableName,
+        axiosTimeoutSeconds,
+        log,
+      });
+
+      const formattedXML = formatXML(res.data);
+      const fileName = createFileName({ cxId, patientId, resources, dateFrom, dateTo });
+
+      await s3Client
+        .putObject({
+          Bucket: bucketName,
+          Key: fileName,
+          Body: formattedXML,
+          ContentType: "application/xml",
+        })
+        .promise();
+
+      const convertUrl = await convertDoc({ fileName, conversionType });
+
+      return convertUrl;
+    } catch (err) {
+      console.log(
+        `Error processing bundle for patient: ${patientId} with resources ${resources}; ${err}`
+      );
+      capture.error(err, {
+        extra: {
+          patientId,
+          resources,
+          dateFrom,
+          dateTo,
+          context: lambdaName,
+        },
+      });
+      throw err;
+    }
+  }
+);
+
+const formatXML = (xml: string): string => {
+  const parser = new DOMParser();
+  const document = parser.parseFromString(xml, "text/xml");
+
+  const structuredBody = document.getElementsByTagName("structuredBody")[0];
+  const components = Array.from(structuredBody.getElementsByTagName("component"));
+
+  const problemListComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Problem list";
+  });
+
+  const treatmentPlanComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Treatment Plan";
+  });
+
+  const medicationComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "History of Medication use";
+  });
+
+  if (problemListComponent) {
+    const problemListTable = problemListComponent?.getElementsByTagName("table")[0];
+
+    addIcd10Codes(problemListComponent, problemListTable, document);
+  }
+
+  if (treatmentPlanComponent) {
+    const firstComponent = components[0];
+    structuredBody.insertBefore(treatmentPlanComponent, firstComponent);
+  }
+
+  if (problemListComponent) {
```

Comment:
Does the code in this `if` need to be executed after the `if (treatmentPlanComponent) {`? If not, it would be easier to read the code if all logic that depends on `problemListComponent` were on the same block.

Reply:
So i put it there to make it easy to insert those components in order